02-feature-engineering.Rmd

```diff
@@ -371,21 +371,19 @@ Removing near-zero variance dummy variables can be a bit tricky because they wil
+If we look closely at our statewide testing data, we will see that there is a considerable amount of missingness. In fact, every row of the data frame has at least one value that is missing. The amount to which missing data impacts your work varies by field, but in most fields you're likely to run into situations where you have to handle missing data in some way. The purpose of this section is to discuss a few approaches using **{recipes}** to handling missingness for predictive modeling purposes. Note that this is not a comprehensive discussion on the topic (for which, we recommend [Little and Rubin (2002)](https://onlinelibrary.wiley.com/doi/book/10.1002/9781119013563)), but is instead an applied discussion of what you *can* do. As with many aspects of data analysis, generally, there is no single approach that will always work best, and it's worth trying a few different approaches in your model development to see how different choices impact your model performance.
 
 There are three basic ways of handling missing data:
 
 -   **Omit** rows of the data frame that include missing values
 -   **Encode** or **Impute** the missing data
 -   **Ignore** the missing data and estimate from the available data
 
-The last option is not always feasible and will depend the modeling framework you're working within. Some estimation procedures can also lend themselves to efficient handling of missing data (for example, imputation via the posterior distribution with Bayesian estimation). In this section, we'll mostly focus on the first three. Additionally, we will only really be concerned with missingness on the predictor variables here, rather than the outcome. Generally, missing data on the predictors is a much more difficult problem than missingness on the outcome, because most models assume you have complete data across your predictors.
+The last option is not always feasible and will depend the modeling framework you're working within. Some estimation procedures can also lend themselves to efficient handling of missing data (for example, imputation via the posterior distribution with Bayesian estimation). In this section, we'll mostly focus on the first two approaches. Additionally, we will only really be concerned with missingness on the predictor variables here, rather than the outcome. Generally, missing data in the predictors is a much more difficult problem than missingness in the outcome, because most models assume you have complete data across your predictors.
 
-### Missing data via {recipes}
```

Comment:
I got rid of this level 3 header bc it seemed redundant, and changed the level 4 headers to level 3 headers, which I think works a bit better. But let me know if you'd like me to change this back

Reply:
Works for me! Thanks for all your work on this today. I know it's a fair amount to go through, but I really appreciate it.

Reply:
of course! it's been fun to read 🙂